posthog/cdp/templates/snapchat_ads/template_snapchat_ads.py

```diff
@@ -0,0 +1,141 @@
+from posthog.cdp.templates.hog_function_template import HogFunctionTemplate
+
+template: HogFunctionTemplate = HogFunctionTemplate(
+    status="alpha",
+    type="destination",
+    id="template-snapchat-ads",
+    name="Snapchat Ads Conversions",
+    description="Send conversion events to Snapchat Ads",
+    icon_url="/static/services/snapchat-ads.png",
+    category=["Advertisement"],
+    hog="""
+let body := {
+    'data': [
+        {
+            'event_name': inputs.eventName,
+            'action_source': inputs.actionSource,
+            'event_time': inputs.eventTime,
+            'user_data': {},
+            'custom_data': {}
+        }
+    ]
+}
+
+if (not empty(event.properties.$current_url)) body.data.1.event_source_url := event.properties.$current_url
+
+for (let key, value in inputs.userData) {
+    // e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 is an empty string hashed
+    if (not empty(value) and value != 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855') {
+        body.data.1.user_data[key] := value
+    }
+}
+
+for (let key, value in inputs.customData) {
+    // e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 is an empty string hashed
+    if (not empty(value) and value != 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855') {
```

Comment:
Really don't like this hack... would love to have this solved differently at some point